### PR TITLE
drivers: can: extend support for automatic sample point location

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Extend support for automatic sample point location to also cover `can_calc_timing()` and `can_calc_timing_data()`. Previously, automatic sample point location was only supported by `can_set_bitrate()` and `can_set_bitrate_data()`.

### CODE REVIEW

File: drivers/can/can_common.c [hunk 1]
```diff
@@ -160,6 +160,30 @@ static int update_sampling_pnt(uint32_t total_tq, uint32_t sample_pnt,
 		sample_pnt - sample_pnt_res;
 }
 
+/**
+ * @brief Get the sample point location for a given bitrate
+ *
+ * @param  bitrate The bitrate in bits/second.
+ * @return The sample point in permille.
```
Comment (kartben) [inline drivers/can/can_common.c:167]: fwiw the inconsistent spelling (permill/permille) is triggering me ;)

Comment (henrikbrixandersen) [reply drivers/can/can_common.c:167]: I had not spotted those, thanks. I've changed them all to "permille".

**Review by alexanderwachter — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
Rebased to resolve merge conflict in `doc/releases/migration-guide-3.7.rst`.

**Review by kartben — DISMISSED** [review]

**Review by alexanderwachter — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
Rebased to resolve merge conflict caused by https://github.com/zephyrproject-rtos/zephyr/pull/69756.

**Review by alexanderwachter — APPROVED** [review]

**Review by str4t0m — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**